sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobOutputStream.java

```diff
@@ -269,7 +269,14 @@ void commit() {
         @Override
         protected void writeInternal(final byte[] data, int offset, int length) {
             this.checkStreamState();
-            sink.next(ByteBuffer.wrap(data, offset, length));
+            /*
+            We need to do a deep copy here because the writing is async in this case. It is a common pattern for
+            customers writing to an output stream to perform the writes in a tight loop with a reused buffer. This
+            coupled with async network behavior can result in the data being overwritten as the buffer is reused.
+             */
+            byte[] buffer = new byte[length];
+            if (length >= 0) System.arraycopy(data, offset, buffer, 0, length);
```

Comment:
nit: this check is kind of redundant. this code would throw line above if value is negative.
is there any validation in the callers of this method that would check length anyway? or maybe we just just throw assertion at the beginning of this method - I think we should throw on negative lenght somewhere.

Reply:
The caller does check this, so I'll remove it here.